### PR TITLE
Make cookbook compatible with ES 7.X

### PR DESCRIPTION
**Pull request by galuhest**
Update the cookbook to be compatible with both ES 6.x and 7.x

### CODE REVIEW

File: recipes/elasticsearch_configure_xpack.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
+version = node['elasticsearch']['version']
+ca = node['elasticsearch']['ca']
+key_name = node['elasticsearch']['xpack_security_transport_ssl_keystore_path']
+xpack_enabled = node['elasticsearch']['xpack_security_enabled']
+bootstrap_password = node['elasticsearch']['bootstrap_password']
+
+if xpack_enabled && version >= '7.0.0' && version < '8.0.0'
```
Comment (ekarisky) [inline recipes/elasticsearch_configure_xpack.rb:7]: are we sure es 8 is different?

File: recipes/elasticsearch_set_index_settings.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+#
+#
+
+port = node['elasticsearch']['port']
+ipaddress = node['ipaddress']
+xpack_enabled = node['elasticsearch']['xpack_security_enabled']
+number_of_shards = node['elasticsearch']['index_number_of_shards']
+number_of_replicas = node['elasticsearch']['index_number_of_replicas']
+refresh_interval = node['elasticsearch']['index_refresh_interval']
+bootstrap_password = node['elasticsearch']['bootstrap_password']
```
Comment (ekarisky) [inline recipes/elasticsearch_set_index_settings.rb]: it is better move this in user side, user need to supplied value with encoded password already, and code only need to decode

**Review by ekarisky — CHANGES_REQUESTED** [review]

**Review by tarabaskara — APPROVED** [review]
if mas @ekarisky approves

File: recipes/elasticsearch_configure_xpack.rb (hunk 1 above)
Comment (galuhest) [reply recipes/elasticsearch_configure_xpack.rb:7]: we don't know yet, that's why we don't set a default value for es8.x and user must set them if they choose to use es8.x

File: recipes/elasticsearch_set_index_settings.rb (hunk 2 above)
Comment (galuhest) [reply recipes/elasticsearch_set_index_settings.rb]: so it's better to not set a default password?

Comment (ekarisky) [reply recipes/elasticsearch_set_index_settings.rb]: yes, please remove default password, user may not need password.

Comment (galuhest) [reply recipes/elasticsearch_set_index_settings.rb]: done, i removed the default bootstrap_password and change the recipe to decode base64 value from user

**Review by ekarisky — APPROVED** [review]

**Pull request merged**